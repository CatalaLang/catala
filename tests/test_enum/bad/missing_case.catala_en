## Article

```catala
declaration enumeration E:
  -- Case1 content integer
  -- Case2 content boolean
  -- Case3

declaration scope A:
  input e content E
  output out content boolean

scope A:
  definition out equals match e with pattern
    -- Case1 of i : i = 0
    -- Case2 of b : b
```

```catala-test-inline
$ catala Interpret -s A
[WARNING] The constructor "Case3" of enumeration "E" is never used; maybe it's unnecessary?

┌─⯈ tests/test_enum/bad/missing_case.catala_en:7.6-7.11:
└─┐
7 │   -- Case3
  │      ‾‾‾‾‾
  └─ Article
[ERROR]
The constructor Case3 of enum E is missing from this pattern matching

┌─⯈ tests/test_enum/bad/missing_case.catala_en:14.25-16.22:
└──┐
14 │   definition out equals match e with pattern
   │                         ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
15 │     -- Case1 of i : i = 0
   │     ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
16 │     -- Case2 of b : b
   │     ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
   └─ Article
#return code 123#
```
