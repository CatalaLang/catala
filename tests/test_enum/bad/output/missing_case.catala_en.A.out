[ERROR] The constructor Case3 of enum E is missing from this pattern matching
[ERROR] 
[ERROR]   --> test_enum/bad/missing_case.catala_en
[ERROR]    | 
[ERROR] 14 |   definition out equals match e with pattern
[ERROR]    |                         ^^^^^^^^^^^^^^^^^^^
[ERROR] 15 |     -- Case1 of i : i = 0
[ERROR]    |     ^^^^^^^^^^^^^^^^^^^^^
[ERROR] 16 |     -- Case2 of b : b
[ERROR]    |     ^^^^^^^^^^^^^^^^^^
[ERROR]    + Article
